config/initializers/rack_attack.rb

```diff
@@ -62,6 +62,12 @@ def remote_ip
   path = request.fullpath
   user = request.env["warden"]&.user
 
+  unless user
+    bearer_token = request.authorization.to_s.split("Bearer ").last
```

Comment:
we're adding an extra lookup to all throttled requests, I wonder if we can return early if there is no bearer token to avoid looking up the API token if it's empty anyway?

Reply:
Updated code to check if token exists 👍